src/peer_transport/handler.ts

```diff
@@ -0,0 +1,151 @@
+import type { IncomingStreamData } from '@libp2p/interface-registrar'
+import { pbStream } from 'it-pb-stream'
+import pDefer, { type DeferredPromise } from 'p-defer'
+import { TimeoutController } from 'timeout-abort-controller'
+import { readCandidatesUntilConnected } from './util'
```

Comment:
Shouldn't this end in `.js`? Otherwise this will fail to import in Node. Similar
comments for other imports.